addons/mail/static/src/core/common/thread_service.js

```diff
@@ -1156,6 +1156,22 @@ export class ThreadService {
             }
         }
     }
+
+    /**
+     * @param {string} searchTerm
+     * @param {Thread} thread
+     */
+    async search(searchTerm, thread) {
+        const messages = await this.rpc("/mail/thread/search", {
+            search_term: searchTerm,
+            thread_id: thread?.id,
+            thread_model: thread?.model,
+        });
+        return messages.map((message) => {
+            message.body = markup(message.body);
+            return this.messageService.insert(message);
+        });
```

Comment:
Ok for a 1st draft version. Should fetch at most 30 messages, and have load more on scroll (or button click).
You don't wanna fetch thousands of messages that match the search term 😅

Reply:
I implemented an infinity scroll for this :)